go/host/host.go

```diff
@@ -875,7 +875,7 @@ func (h *host) handleBatches(encodedBatchMsg *common.EncodedBatchMsg) error {
 
 	for _, batch := range batchMsg.Batches {
 		// The enclave retrieves the genesis from the L1 chain, so we do not need to submit it.
-		if batch.Header.Number == big.NewInt(int64(common.L2GenesisHeight)) {
+		if batch.Header.Number.Cmp(big.NewInt(int64(common.L2GenesisHeight))) == 0 {
```

Comment:
This meant that we'd never store the genesis batch, because we were doing the compare wrong.